app/scripts/lib/ppom/ppom-middleware.ts

```diff
@@ -81,13 +79,9 @@ export function createPPOMMiddleware<
       const securityAlertsEnabled =
         preferencesController.store.getState()?.securityAlertsEnabled;
 
-      // This will always exist as the SelectedNetworkMiddleware
-      // adds networkClientId to the request before this middleware runs
-      const { chainId } =
-        // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
-        networkController.getNetworkConfigurationByNetworkClientId(
-          req.networkClientId,
-        )!;
+      const { chainId } = getProviderConfig({
+        metamask: networkController.state,
+      });
```

Comment:
This now needs to handle cases where result from `getProviderConfig` is undefined.

https://app.circleci.com/pipelines/github/MetaMask/metamask-extension/103365/workflows/9cc2028a-dd02-4be0-810c-02a3872f87e3/jobs/3852684

```
app/scripts/lib/ppom/ppom-middleware.ts:82:15 - error TS2339: Property 'chainId' does not exist on type '{ id?: string | undefined; nickname?: string | undefined; rpcUrl?: string | undefined; chainId: `0x${string}`; ticker: string; rpcPrefs: any; type: string; } | undefined'.
```

Reply:
resolved here https://github.com/MetaMask/metamask-extension/pull/27570/commits/06facfad2d4b6e9ea56d0ddea74dd6c1eb1710ca